Source/WebCore/xml/parser/XMLDocumentParserLibxml2.cpp

```diff
@@ -466,11 +467,14 @@ static bool shouldAllowExternalLoad(const URL& url)
 
 static void* openFunc(const char* uri)
 {
-    ASSERT(XMLDocumentParserScope::currentCachedResourceLoader);
+    ASSERT(XMLDocumentParserScope::currentCachedResourceLoader());
     ASSERT(libxmlLoaderThread == &Thread::current());
 
-    CachedResourceLoader& cachedResourceLoader = *XMLDocumentParserScope::currentCachedResourceLoader;
-    Document* document = cachedResourceLoader.document();
+    RefPtr cachedResourceLoader = XMLDocumentParserScope::currentCachedResourceLoader().get();
+    if (!cachedResourceLoader)
+        return &globalDescriptor;
+
+    RefPtr document = cachedResourceLoader->protectedDocument();
```

Comment:
`RefPtr document = cachedResourceLoader->document();`

Reply:
I'm really curious why `document()` is preferred over `protectedDocument()` here.

I guess I don't understand the purpose of `protectedDocument()` if it's not used here.  Is it only supposed to be used internal to the class itself?

Reply:
Ah, I see this is in the code guidelines.  The `protectedFoo()` functions are to avoid a local variable.

Reply:
Calling `protectedDocument()` here doesn't hurt but it also has no benefit over calling `document()` since you're storing it in a RefPtr anyway. No benefit, not even perf. However, it is more verbose so there is no reason to change existing code.

`protectedDocument()` is indeed useful to avoid a local variable, e.g.:
```
protectedDocument()->doSomething();
```

Reply:
Oops, meant to push a fix to replace `protectedDocument()` before landing this.  Will fix in a moment.

Reply:
Follow-up PR:  https://github.com/WebKit/WebKit/pull/27724